src/components/composer/composer/abstract_composer_store.ts

```diff
@@ -556,6 +559,51 @@ export abstract class AbstractComposerStore extends SpreadsheetStore {
     return highlight && highlight.color ? highlight.color : undefined;
   }
 
+  private computeFormulaCursorContext() {
+    if (this.selectionStart === this.selectionEnd) {
+      const parenthesesCodeAtCursor = this.getParenthesesCodeAfterCursor();
+      const previousSymbolAtCursor = [...this.currentTokens]
+        .reverse()
+        .find((t) => parenthesesCodeAtCursor.startsWith(t.parenthesesCode!) && t.type === "SYMBOL");
+
+      if (previousSymbolAtCursor) {
+        for (let i = 0; i < this.currentTokens.length; i++) {
+          this.currentTokens[i].isFormulaCursorContextToken = this.isTokenInCursorFormulaContext(
+            this.currentTokens[i],
+            previousSymbolAtCursor
+          );
+        }
+        return;
+      }
+    }
+
+    for (let i = 0; i < this.currentTokens.length; i++) {
+      this.currentTokens[i].isFormulaCursorContextToken = true;
+    }
+  }
+
+  private getParenthesesCodeAfterCursor(): string {
+    // we always look at the code associated with the token located after the cursor.
+    // This code is the same as the 'tokenAtCursor' except in the case of a closing parenthesis.
+    // In this case we look at the code located one degree below in the parentheses tree
+    const code = this.tokenAtCursor?.parenthesesCode || "";
+    if (this.tokenAtCursor?.type === "RIGHT_PAREN") {
+      return code.slice(0, -1) || "";
+    }
+    return code;
+  }
+
+  private isTokenInCursorFormulaContext(
+    token: EnrichedToken,
+    // we refer to the previous symbol parenthesesCode and not directly the
+    // parenthesesCode of the token at the cursor because we don't want to
+    // match cases where the token at the cursor is between parentheses which
+    // are not function parentheses
```

Comment:
Why shoudn't we highlight the content of the parenthesis if it isn't a function ? It make sense no ?

Reply:
and the function could probably be inlined

Reply:
I will say that 

1) It is important for the user to always have visual information on which sub-formula he is editing. 
Particularly when he is trying to edit a formula, and he moves his cursor inside it

2) That in the case where he edits a mathematical operation composed of several parentheses, then he can easily move his cursor to the nearest parenthesis to have another visual information on the corresponding matching parenthesis

inlined cases are taken into account